File: Python/backend_flask/shared/dto/schemas/user_dto.py
user_dto = {
    "type": "object",
    "properties": {
        "email": {"type": "string", "format": "email"},
        "first_name": {"type": "string"},
        "last_name": {"type": "string"},
        "role": {"type": "string", "enum": ["admin", "reader"]},
    },
    "required": ["email", "first_name", "last_name", "role"],
    "additionalProperties": False
}

File: Javascript/backend_nest/src/dto/schema/UserDTO.ts
import { IsString, IsEmail, MaxLength, IsOptional } from 'class-validator';

export class UserDTO {
  @IsOptional()
  id?: number;

  @IsEmail()
  email: string;

  @IsString()
  @MaxLength(100)
  firstName: string;

  @IsString()
  @MaxLength(100)
  lastName: string;

  @IsString()
  role: string;
}
